chain/yoda/context.go

```diff
@@ -47,6 +47,10 @@ type Context struct {
 
 	dataSourceCache *sync.Map
 	pendingRequests map[types.RequestID]bool
+
+	handlingGauge  int64
+	pendingGauge   int64
+	submittedCount int64
 }
 
```

Comment:
I think create increase function for each variable (increaseHandlingGauge, increasependingGauge, increasesubmittedCount) and update when flag is set(dont' do the atomic add if other node don't run the prometheus)